var/spack/repos/builtin/packages/bcl2fastq2/package.py

```diff
@@ -27,7 +27,7 @@ class Bcl2fastq2(Package):
     conflicts('platform=darwin',
               msg='malloc.h/etc requirements break build on macs')
 
-    depends_on('boost@1.54.0')
+    depends_on('boost@1.55.0')
```

Comment:
```suggestion
    depends_on('boost@1.55.0:')
```

Does the latest version work? The above syntax just adds a lower-bound, and no upperbound. If that does not work, a better idea would be to bisect the boost versions 1.53.0 to 1.77.0 until you reach something that *does* compile (that would require 4 to 5 builds...), and then replace this with `depends_on('boost@1.54.0:1.xx.0')` where `xx` is a known working version. That way we don't force a specific boost version and the user could reuse whatever they already have installed through spack.

Reply:
I tried higher-numbered versions of boost, but those introduced further compile errors in the Logger functions. While workarounds for those exist (see [https://www.biostars.org/p/387127/#394722](https://www.biostars.org/p/387127/#394722) for example, Illumina has introduced a new tool caled [BCL Convert](https://support.illumina.com/sequencing/sequencing_software/bcl-convert.html) that they are recommending as a replacement for bcl2fastq2. Our scientists have not yet adopted BCL Convert, which requires some [significant changes to use](https://support.illumina.com/bulletins/2020/10/upgrading-from-bcl2fastq-to-bcl-convert.html) so in this case since changing boost from 1.54.0 to 1.55.0 doesn't break anything, I'd like to stick with this minor update.

Reply:
Ok, sounds good to me,

Reply:
With #26659 merged, `bcl2fastq2` can build with `boost@1.54` (it is the minimum supported version) and `1.55` (I confirm 1.55 is the newest version for which no changes in bsc2fastq are needed)

bcl2fastq2 is not mainained anymore and it's last release (which we have) is from 2017.

Reply:
```suggestion
    depends_on('boost@1.54.0:1.55')
```
You can apply this suggestion as well.
1.55 is fine in any case and 1.54 as well when @26659 is merged.

Reply:
@alansill I hope you'd be ok if I apply my suggestion to allow the build with:
* 1.55 (default, as it is the newer version) and optionally also
* 1.54 (when selected by spack or by `spack install bcl2fastq2 ^boost@1.54.0`)
** assuming that #26659 is merged soon a well